userspace/libscap/scap_savefile.c

```diff
@@ -1164,6 +1171,48 @@ static int32_t scap_read_proclist(scap_t *handle, gzFile f, uint32_t block_lengt
 
 		totreadsize += readsize;
 
+		switch(block_type)
+		{
+		case PL_BLOCK_TYPE_V1:
+		case PL_BLOCK_TYPE_V1_INT:
+		case PL_BLOCK_TYPE_V2:
+		case PL_BLOCK_TYPE_V2_INT:
+		case PL_BLOCK_TYPE_V3:
+		case PL_BLOCK_TYPE_V3_INT:
+		case PL_BLOCK_TYPE_V4:
+		case PL_BLOCK_TYPE_V5:
+		case PL_BLOCK_TYPE_V6:
+			break;
+		case PL_BLOCK_TYPE_V7:
+			//
+			// exepath
+			//
+			readsize = gzread(f, &(stlen), sizeof(uint16_t));
+			CHECK_READ_SIZE(readsize, sizeof(uint16_t));
+
+			if(stlen > SCAP_MAX_PATH_SIZE)

```

Comment:
I think this should be `stlen >= SCAP_MAX_PATH_SIZE`, otherwise when you do `tinfo.exepath[stlen] = 0;` some lines below, you risk to do an off-by-one.